src/NetworkDynamics.jl

```diff
@@ -44,20 +44,20 @@ function collect_ve_info(vertices!, edges!, graph)
         symbols_e = [Symbol(edges![i].sym[j], "_", i)
                      for i in 1:length(edges!)
                      for j in 1:e_dims[i]]
-        if eltype(edges!) <: Union{StaticEdge,StaticDelayEdge}  # improve type hierarchy
-            mme_array = nothing
-        else
+        if eltype(edges!)<:ODEEdge #at the moment 1 ODEEdge should imply all e are ODEEdge
             mme_array = [e.mass_matrix for e in edges!]
+        else
+            mme_array = nothing
```

Comment:
Actually I prefer the old way. Currently there is nothing preventing `edges! <: Vector{EdgeFunction}` (i.e. mixed types).
In that case the old code would error. The new code simply drops all mass matrices. This won't happen currently but might cause silent bugs in the future if someone decides to fiddle with the edge preparation.

alternatively one could add
```
@assert isconcretetype(eltype(edges!)) # i think thats the function but haven't checked
```

Reply:
Mixed types (Static + StaticDelay) are fine as long as there is no ODEEdge mixed in. Only those have mass matrices. 
This line in `network_dynamics` should prevent a mix with ODEEdges from occuring:
```
    if hasODEEdge && hasStaticEdge
        edges! = Array{ODEEdge}(edges!)
    end
```

I changed the line because the previous version lead to an error with mixed types.

Reply:
I was also thinking about introducing another abstract supertype for (StaticEdge and StaticDelayEdge) but it seemed to be a bit of an overkill.

Reply:
New solution: `hasMM = any(e -> hasproperty(e, :mass_matrix), edges!)`